hardware_frame_context.go

```diff
@@ -0,0 +1,48 @@
+package astiav
+
+//#include <libavcodec/avcodec.h>
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+type HardwareFrameContext struct {
+	c *C.struct_AVBufferRef
+}
+
+func AllocHardwareFrameContext(hdc *HardwareDeviceContext) (*HardwareFrameContext, error) {
+	if hfc := C.av_hwframe_ctx_alloc(hdc.c); hfc != nil {
```

Comment:
Can you create a `newHardwareFrameContextFromC` func similar to [this](https://github.com/asticode/go-astiav/blob/master/output_format.go#L13) and use it here instead of `&HardwareFrameContext{c: hfc}`?